### PR TITLE
TF-495 Apply Isolate for LoadMore use-case

### PR DESCRIPTION
**Pull request by dab246**
### The problem is solved in this PR
- [x] Apply Isolate for LoadMore use-case
   - Increase email loading speed avoid blocking user interaction and other tasks.
- [x] Integration `linter` plugin
   - To encourage good coding practices.
   - Performance improvement
 
278493242_5049713618455851_132285591329688861_n

### Result

- The app stays at ` 60 fps` most of the time, or in other words, changing frames every `16 milliseconds`.
- Device Test: Oppo - Android 10

278781581_4827297477367408_1328088794709915929_n

### CODE REVIEW

**Comment by hoangdat** [comment]
@dab246 please add the image for the `perfomance_overlay` before apply your solution.

And also please add more about your device: RAM, ROM, refresh rate :)

**Comment by dab246** [comment]
> @dab246 Great! please add the image for the `perfomance_overlay` before apply your solution.
> 
> And also please add more about your device: RAM, ROM, refresh rate :)
 
I updated

**Review by hoangdat — APPROVED** [review]

File: lib/features/thread/presentation/thread_controller.dart [hunk 1]
```diff
@@ -382,13 +387,15 @@ class ThreadController extends BaseController {
   void loadMoreEmails() {
     if (canLoadMore && _accountId != null) {
       log('ThreadController::loadMoreEmails(): latest: ${emailList.last.receivedAt}');
+      startFpsMeter();
```
Comment (hoangdat) [inline lib/features/thread/presentation/thread_controller.dart:390]: how to only apply it on debug or profile mode?

File: core/lib/utils/benchmark.dart [hunk 2]
```diff
@@ -0,0 +1,29 @@
+
+import 'package:core/utils/app_logger.dart';
+
+class _Benchmark {
```
Comment (hoangdat) [inline core/lib/utils/benchmark.dart:4]: should run it only in debug or profile mode

Comment (dab246) [reply core/lib/utils/benchmark.dart:4]: oke i will update it

File: lib/features/thread/presentation/thread_controller.dart (hunk 1 above)
Comment (dab246) [reply lib/features/thread/presentation/thread_controller.dart:390]: Only check `Foundation.kDebugMode` or `Foundation.kProfileMode`

Comment (hoangdat) [reply lib/features/thread/presentation/thread_controller.dart:390]: yes, please apply it from the Fps class, or here, or inside the startFpsMeter()

**Review by hoangdat — APPROVED** [review]

**Comment by hoangdat** [comment]
image

**Pull request merged**